src/transformers/models/transfo_xl/modeling_transfo_xl.py

```diff
@@ -1097,16 +1097,30 @@ def forward(
         last_hidden = transformer_outputs[0]
         pred_hid = last_hidden[:, -tgt_len:]
 
+        if labels is not None:
+            # Prevents all labels being -100 and throwing an error
+            # when backwarding the loss
+            miss_valid_label = labels[0, 1:].sum() == (labels.size(1) - 1) * -100
+            if miss_valid_label:
+                # Sets an <EOS> token, just to prevent loss from being NaN
+                labels[0, 1] = self.config.eos_token_id
+
         softmax_output = self.crit(pred_hid, labels)
         prediction_scores = softmax_output.view(bsz, tgt_len, -1) if labels is None else ()
-        loss = softmax_output.view(bsz, tgt_len - 1) if labels is not None else None
+
+        if labels is not None:
+            loss = softmax_output.view(bsz, tgt_len - 1)
+            # Avoids from incorporating padding (-100) tokens into loss value
+            loss = loss[loss != 0].mean()
```

Comment:
here we go from a a vector to a scalar which is also very much a breaking change. Why do we need to do this?

Reply:
I would say that since we are using a customizable loss layer, which is the ProjectiveAdaptiveLogSoftmax (https://github.com/huggingface/transformers/blob/master/src/transformers/models/transfo_xl/modeling_transfo_xl_utilities.py#L29) and it outputs a vector instead of a scalar, we are unable to perform the backward pass inside the Trainer API when it comes to the time (https://github.com/huggingface/transformers/blob/master/src/transformers/trainer.py#L1885).

If we gather GPT2 as an example, its loss is already reduced when it comes out of the CrossEntropyLoss (https://github.com/huggingface/transformers/blob/master/src/transformers/models/gpt2/modeling_gpt2.py#L1075).